tensorflow/core/ops/mkl_nn_ops.cc

```diff
@@ -72,7 +72,7 @@ REGISTER_OP("_MklFusedMatMul")
     .Attr("is_filter_const: bool = false")
     .Attr("transpose_a: bool = false")
     .Attr("transpose_b: bool = false")
-    .Attr("T: {float}")
+    .Attr("T: {bfloat16, float}")
```

Comment:
(Quoted the wrong side. Sorry. I meant the new code. Reposting the comment.)
Should we guard this with `#ifdef ENABLE_INTEL_MKL_BFLOAT16`? Should `_MklFusedMatMul` still take just `float` if `ENABLE_INTEL_MKL_BFLOAT16` is not defined?

Reply:
We consider such strategy:
* because current Eigen may not support BF16 well, only enable BF16 in Eigen when user really wants to use(guard Eigen ops with `ENABLE_INTEL_MKL_BFLOAT16`)
* MKLDNN has suppported BF16 well in its kernel, we can enable it by default without risk.

If this strategy makes confusion, I will guard all bfloat16 definition with `ENABLE_INTEL_MKL_BFLOAT16`. How do you think about this?

Reply:
I think this is fine then. Thank you for the clarification!